dockstore-client/src/main/java/io/github/collaboratory/wdl/WDLClient.java

```diff
@@ -198,35 +205,58 @@ public Boolean check(File content) {
                     outputFound = true;
                 }
             }
-            //check all the required fields and give error message if it's missing
-            if (counter > 0 && wfFound && commandFound && callFound && outputFound) {
-                return true;    //this is a valid WDL file
+            // check all the required fields and give error message if it's missing
+            if (wfFound && callFound) {
+                String missingPotentiallyRequiredFields = "";
+                if (counter == 0) {
+                    missingPotentiallyRequiredFields += " '" + TASK + "'";
+                }
+                if (!commandFound) {
+                    missingPotentiallyRequiredFields += " '" + COMMAND + "'";
+                }
+                if (!outputFound) {
+                    missingPotentiallyRequiredFields += " '" + OUTPUT + "'";
+                }
+
+                if (!missingPotentiallyRequiredFields.isEmpty()) {
+                    // Gives user a warning if their entry file doesn't contain task, call, or output
+                    out(WDL_CHECK_WARNING_MESSAGE + missingPotentiallyRequiredFields);
+                }
+
+                return true;    // this is potentially valid WDL file
             } else if (counter == 0 && !wfFound && !commandFound && !callFound && !outputFound) {
-                return false;   //not a WDL file, maybe a CWL file or something else
+                return false;   // not a WDL file, maybe a CWL file or something else
             } else {
-                //WDL file but some required fields are missing
+                // WDL file but some required fields are missing
+                String missingPotentiallyRequiredFields = "";
```

Comment:
Could use `List` as well here and next line.

Reply:
Fixed!